blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/QueuedBlueRun.java

```diff
@@ -0,0 +1,159 @@
+package io.jenkins.blueocean.service.embedded.rest;
+
+import com.google.common.collect.ImmutableList;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.hal.Links;
+import io.jenkins.blueocean.rest.model.BlueActionProxy;
+import io.jenkins.blueocean.rest.model.BlueArtifactContainer;
+import io.jenkins.blueocean.rest.model.BlueChangeSetEntry;
+import io.jenkins.blueocean.rest.model.BluePipelineNodeContainer;
+import io.jenkins.blueocean.rest.model.BluePipelineStepContainer;
+import io.jenkins.blueocean.rest.model.BlueRun;
+import io.jenkins.blueocean.rest.model.Container;
+import org.kohsuke.stapler.export.Exported;
+
+import java.util.Collection;
+import java.util.Date;
+
+public class QueuedBlueRun extends BlueRun {
+
+    public static final String QUEUE_ID = "queueId";
+
+    private final BlueRunState runState;
+    private final BlueRunResult runResult;
+    private final QueueItemImpl item;
+    private final Link parent;
+
+    public QueuedBlueRun(BlueRunState runState, BlueRunResult runResult, QueueItemImpl item, Link parent) {
+        this.runState = runState;
+        this.runResult = runResult;
+        this.item = item;
+        this.parent = parent;
+    }
+
+    @Override
+    public String getOrganization() {
+        return item.getOrganization();
+    }
+
+    @Override
+    public String getId() {
+        return Integer.toString(item.getExpectedBuildNumber());
+    }
+
+    @Exported(name = QUEUE_ID)
+    public String getQueueId() {
```

Comment:
@vivek to allay your concern the client can always find out if the Run is actually a queued run by the presence of this property in the fetch response.